src/Glpi/Api/HL/Router.php

```diff
@@ -619,17 +621,20 @@ public function handleRequest(Request $request): Response
             }
         }
 
-        try {
-            $this->handleAuth($request);
-        } catch (OAuthServerException $e) {
-            return new JSONResponse(
-                content: AbstractController::getErrorResponseBody(
-                    status: AbstractController::ERROR_INVALID_PARAMETER,
-                    title: 'Invalid OAuth token',
-                    detail: $e->getHint()
-                ),
-                status: 400
-            );
+        if ($CFG_GLPI['enable_hlapi']) {
+            // OAuth will only be used if the API is enabled
+            try {
+                $this->handleAuth($request);
+            } catch (OAuthServerException $e) {
+                return new JSONResponse(
+                    content: AbstractController::getErrorResponseBody(
+                        status: AbstractController::ERROR_INVALID_PARAMETER,
+                        title: 'Invalid OAuth token',
+                        detail: $e->getHint()
+                    ),
+                    status: 400
+                );
+            }
```

Comment:
Will this prevent the inventory agent to be able to use the oauth authentication?
Maybe we should have 2 distinct configs:
 - `enable_hlapi`: to enable access to HL API endpoints from a web request;
 - `enable_oauth_server`: to enable the Oauth server to be used (by the inventory agent for instance).
The `enable_oauth_server` config could be enforced if the HL API is enabled.